### ISSUE TITLE
BlobStorageResource not uploading files

### ISSUE THREAD

**Issue by przem** [microsoft/spring-cloud-azure#667]
BlobStorageResource is somehow silently failing on upload files on azure storage. 
It creates a container when needed, but does not upload a file, despite that autoCreateFiles is set to true.

If file is manually created, then you get exception that file already exists, so no overwrite option.

## Code for Reproduce 
```java
String path = String.format("azure-blob://%s/%s", container, location);
BlobStorageResource blobFile = (BlobStorageResource) loader.getResource(path);
IOUtils.copy(inputStream, blobFile.getOutputStream());
```

## Branch
* 1.2.3 version

## Your Environment
* OpenJDK 14
* Spring boot 2.3.0.M4

**Comment by saragluna** [comment]
Thanks for reaching out. I notice you're using Spring Boot 2.3.x, have you tried with Spring Boot 2.2.x cause I'm not sure whether 1.2.3 could work with Spring Boot 2.3.x.

**Comment by przem** [comment]
No I haven't, and its not possible for me to downgrade the project to spring 2.2.x
I might be wrong, but i think spring version is not involved, its simple stream copy to BlobStorageResource.

As a workaround i've overwritten few things here and there to execute:
`this.blockBlobClient.upload(inputStream, size, true);`
instead of 
`return this.blockBlobClient.getBlobOutputStream(true);`

So maybe you could expose this upload method as well on the BlobStorageResource

**Comment by saragluna** [comment]
Sorry for the late response, and as we tested with Spring Boot 2.2.x this code snippet worked. We'll check this with Spring Boot 2.3.0 and see how to fix it.

**Comment by saragluna** [comment]
This issue has been fixed in #752, closing this now.
